File: Problems/range_sum_of_bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sum;
    public int rangeSumBST(TreeNode root, int low, int high) {
        sum = 0;
        func(root,low,high);
        return sum;
    }
    public void func(TreeNode root,int low,int high)
    {
        if(root == null)
        {
            return;
        }
        if(root.val >= low && root.val <= high)
        {
            sum += root.val;
            func(root.left,low,high);
            func(root.right,low,high);
        }
        else
        {
            if(root.val < low)
            {
            func(root.right,low ,high);
            }
            
            if(root.val > high)
            {
            func(root.left, low, high);
            }
        }
        
    }
}

File: Problems/17 July 2021/range_sum_of_bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        sum_val = 0
        if not root:
            return 0
        elif root.val >=low  and root.val <= high:
            sum_val += root.val
        return sum_val + self.rangeSumBST(root.left, low, high) + self.rangeSumBST(root.right, low, high)